### PR TITLE
feat: support a list of rules instead of a single rule

### PR DESCRIPTION
**Pull request by exocom**
## Description

Modify CORS to support multiple rules instead of a single rule
## Motivation and Context

When importing an existing bucket the module was attempting to remove all but one of the rules.
## Breaking Changes

Yes

Object to Array. It is possible to be backwards compatible, however, since the syntax is as simple as adding brackets there is little reason to add additional code.

## How Has This Been Tested?

Applying on an existing resource

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -82,10 +82,16 @@ variable "website" {
   default     = {}
 }
 
-variable "cors_rule" {
-  description = "Map containing a rule of Cross-Origin Resource Sharing."
-  type        = any # should be `map`, but it produces an error "all map elements must have the same type"
-  default     = {}
+variable "cors_rules" {
+  description = "List containing rules for Cross-Origin Resource Sharing."
+  type        = list(object({
```
Comment (bryantbiggs) [inline variables.tf:87]: lets use
```hcl
type = list(any)
```

instead. if we spell out each attribute like below, this breaks the optional arguments and becomes very strict

Comment (exocom) [reply variables.tf]: @bryantbiggs changes have been made.

Personally I'm a fan of the strict, however, I also understand the desire for optional arguments :)

**Comment by exocom** [comment]
@bryantbiggs any chance we can get this merged in?

**Comment by bryantbiggs** [comment]
@antonbabenko :+1:

**Comment by antonbabenko** [comment]
I renamed variable to be the same as before - `cors_rule`. This way users should wrap value to make it `list`.

**Comment by antonbabenko** [comment]
Thanks @exocom for this PR!

[v1.11.0](https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/releases/v1.11.0) has been just released.

**Pull request merged**